### PR TITLE
Auth Modal: Custom modal styles integration

### PR DESCRIPTION
**Pull request by jmderby**
## Description

These changes directly correspond to this [PR](https://github.com/Paella-Labs/crossbit-main/pull/14498).
tldr;
From the demo app, we're passing a custom styling object: `appearance` through the `CrossmintAuthProvider` via props, which then sends a request to open the iframe with the `uiConfig` query param in the request which the backend will interpret and display the styles accordingly.

## Test plan

App flow still works as expected.
The pre-existing auth modal still works as before
Demo of custom demo design:
https://www.loom.com/share/a86664af2d514586a801cb2e21dd2093

Screenshot of default design working as expected:
Screenshot 2024-08-30 at 2 39 23 AM
Screenshot 2024-08-30 at 2 40 49 AM

### CODE REVIEW

File: apps/wallets/smart-wallet/next/src/app/template.tsx [hunk 1]
```diff
@@ -31,6 +31,22 @@ export default function Template({ children }: { children: React.ReactNode }) {
         <CrossmintProvider apiKey={process.env.NEXT_PUBLIC_CROSSMINT_API_KEY ?? ""}>
             <CrossmintAuthProvider
                 embeddedWallets={{ createOnLogin: "all-users", type: "evm-smart-wallet", defaultChain: "polygon-amoy" }}
+                appearance={{
+                    appLogo:
```
Comment (jmderby) [inline apps/wallets/smart-wallet/next/src/app/template.tsx:35]: todo: get appLogo and appName from the crossmint console db

Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:35]: in future PR

File: packages/client/auth-core/src/components/AuthModal.tsx [hunk 2]
```diff
@@ -30,13 +31,20 @@ export default function AuthModal({
...
 }: {
     setModalOpen: (open: boolean) => void;
     setJwtToken: (jwtToken: string) => void;
     apiKey: string;
     baseUrl: string;
+    appearance?: UIConfig;
 }) {
-    const iframeSrc = `${baseUrl}/sdk/auth/frame?apiKey=${apiKey}`;
+    let iframeSrc = `${baseUrl}/sdk/auth/frame?apiKey=${apiKey}`;
+    if (appearance) {
```
Comment (devlyn37) [inline packages/client/auth-core/src/components/AuthModal.tsx:43]: pls use loose null check `appearance != null`, it's a convention at Crossmint!

File: apps/wallets/smart-wallet/next/src/app/template.tsx [hunk 3]
```diff
@@ -31,6 +31,25 @@ export default function Template({ children }: { children: React.ReactNode }) {
...
...814 15.3173 25.2625 16.5588 28.7803C17.8004 32.298 18.4812 34.394 19.6928 35.2083C20.9043 36.0226 23.3073 35.6089 24.0716 34.7812' stroke='%23602C1B' stroke-width='2.5' stroke-linecap='round' stroke-linejoin='round'%3E%3C/path%3E%3Cpath d='M16.5491 23.792C15.7326 22.5723 14.0598 20.439 12.7716 16.9352C11.8009 14.2953 9.73159 13.4674 8.07616 15.5367C6.42074 17.606 8.01611 21.0801 9.49463 23.792C10.9732 26.5039 13.6633 30.2286 14.698 31.2632C15.3187 34.781 14.5083 37.4718 15.9545 39.9509C18.0087 43.472 22.1474 43.8858 26.079 42.8512C28.7691 42.4374 31.0453 44.0928 34.77 43.8858C38.4947 43.6789 39.7363 39.7473 41.3917 37.4711C43.0471 35.1949 45.9441 32.5048 47.1857 31.0563C48.4272 29.6078 48.0134 27.5385 46.5649 26.9178C45.1164 26.297 43.8582 26.5151 42.2194 28.1593C40.6806 29.7032 39.3224 31.884 39.3224 31.884C39.3224 31.884 36.8393 31.4702 34.5631 32.2979C32.0052 33.228 30.0106 34.3672 28.3552 36.8503' stroke='%23602C1B' stroke-width='2.5' stroke-linecap='round' stroke-linejoin='round'%3E%3C/path%3E%3C/g%3E%3C/g%3E%3Cdefs%3E%3CclipPath id='clip0_943_5959'%3E%3Crect width='54' height='54' fill='white' transform='translate(0.660156)'%3E%3C/rect%3E%3C/clipPath%3E%3C/defs%3E%3C/svg%3E",
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/app/template.tsx:37]: nit: Can you move this to another file or something? Feels noisy here

File: apps/wallets/smart-wallet/next/src/app/template.tsx [hunk 4]
```diff
@@ -31,6 +31,25 @@ export default function Template({ children }: { children: React.ReactNode }) {
...
+                    appName: "Nike",
```
Comment (devlyn37) [inline apps/wallets/smart-wallet/next/src/app/template.tsx:38]: Intentional?

**Review by devlyn37 — COMMENTED** [review]
pls add changesets

File: apps/wallets/smart-wallet/next/src/app/template.tsx (hunk 4 above)
Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:38]: fixed

File: apps/wallets/smart-wallet/next/src/app/template.tsx (hunk 1 above)
Comment (jmderby) [reply apps/wallets/smart-wallet/next/src/app/template.tsx:36]: @devlyn37 Created a follow up RE: this
https://linear.app/crossmint/issue/WAL-2810/custom-auth-modal-app-name-and-app-logo

**Review by devlyn37 — APPROVED** [review]
Nice!

**Pull request merged**